### PR TITLE
[Term Entry] D3 Selection .join() added

### LINKED ISSUE #3957: [Term Entry] D3 Selection .join()
### Reason/inspiration (optional)

We would like a new term entry on the `.join()` method for [D3](https://www.codecademy.com/resources/docs/d3): Selection. The entry should go in a new file under `docs/content/d3/concepts/selection/terms/join/join.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```js `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by antriksh-9**
### Description

I was assigned this [issue](https://github.com/Codecademy/docs/issues/3957).
Created Codecademy Docs term entry on the .join() method for D3.
CLoses #3957.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/d3/concepts/selection/terms/join/join.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+The **`.join()`** method in D3.js efficiently binds data, appends new elements for entering data, updates existing elements, and removes elements without corresponding data, returning a merged enter and update selection.
+
+## Syntax
+
+```pseudo
+selection.join(
+  enterFunction,
+  updateFunction,
+  exitFunction
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/join/join.md:25]:
```suggestion
  enter,
  update,
  exit

```
Better to keep it as simple as possible.

File: content/d3/concepts/selection/terms/join/join.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+);
+```
+
+- `enterFunction`:  Function specifying how to create new elements for entering data.
+- `updateFunction`: Function specifying how to update properties of existing elements based on data.
+- `exitFunction`: Function specifying how to handle elements that no longer have corresponding data.
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/join/join.md:31]:
```suggestion
- `enter`:  Function specifying how to create new elements for entering data.
- `update`: Function specifying how to update properties of existing elements based on data.
- `exit`: Function specifying how to handle elements that no longer have corresponding data.

```

File: content/d3/concepts/selection/terms/join/join.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    update => update.text(d => d),            // Update existing elements for update selection
+    exit => exit.remove()                     // Remove elements for exit selection
+  );
+</script>
+
+</body>
+</html>
+```
+
+In this example, if there are fewer `p` elements than data points, new `p` elements are created (enter selection). If there are an equal number, existing elements are updated (update selection). If there are more elements than data points, excess elements are removed (exit selection).
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/join/join.md:64]:
```suggestion
In this example, if there are fewer `p` elements than data points, new `p` elements are created (`enter` selection). If there is an equal number, existing elements are updated (`update` selection). If there are more elements than data points, excess elements are removed (`exit` selection).

```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Excellent Work @antriksh-9, I reviewed the content and mentioned some changes, please make them.

**Comment by antriksh-9** [comment]
Hey @avdhoottt I have made the changes.

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review, Thanks for the changes.

**Review by avdhoottt — APPROVED** [review]
LGTM!

**Pull request merged**